setup.py

```diff
@@ -43,6 +43,10 @@ def set_bin_files(data_files, dest, src=None):
         src = ["bin/waagent", "bin/waagent2.0"]
     data_files.append((dest, src))
 
+def set_bin_version2_files(data_files, dest, src=None):
```

Comment:
we don't need this new function, do we? we can use set_bin_files

Reply:
Idea for separate function is we only set new binary file if that distro has py3 requirement rest we use set_bin_files.

Reply:
Keeping in one place required extra logic and need to know distro type

Reply:
the only thing we are changing is the default value... am i missing something? instead of creating a new function with a different default value you could just call the original function with the different value.

Reply:
and we also need to copy waagent2.0

Reply:
That's is good point but my initial idea is to keep this new path as default value in other function since this is constant path(not change for other distros). This would avoid passing every time. I'm happy to change to use original function.

The current setup is first we are setting old files(waagent & waagent2.0) then overriding the waagent with new file.

Reply:
I think having 2 functions that do the same and only change the default value is very odd, but you can leave as is if you feel strongly about it.
This, though, we should change:

> The current setup is first we are setting old files(waagent & waagent2.0) then overriding the waagent with new file.

so you saying for SLES first we call set_bin_files and then set_bin_version2_files? that is confusing... I guess that is why I missed that waagent2.0 was being copied. Other people reading the code may even think that it is a bug (calling both functions)

Reply:
Not only for SLES.. The setup was codded in way that we set binary files first irrespective of what distro it is.. If we want to call only one time then I would have to remove that and include all 10 if block statements..

Reply:
calling outside https://github.com/Azure/WALinuxAgent/blob/f59375f211951ffe17aea1ff6c1949c7c8d16319/setup.py#L103

Reply:
I see. I looked at the rest of the code and it could be better abstracted, but for now calling set_bin_files with the correct files for each of the distros (just ad we call set_conf_files) would be clearer

Reply:
cool I will update that

Reply:
addressed